core/src/ast/types.rs

```diff
@@ -372,102 +371,44 @@ impl TypeName {
     pub fn to_syn(&self) -> syn::Type {
         match self {
             TypeName::Primitive(name) => {
-                syn::Type::Path(syn::parse_str(PRIMITIVE_TO_STRING.get(name).unwrap()).unwrap())
+                let ident = name.to_ident();
+                syn::parse_quote_spanned!(Span::call_site() => #ident)
```

Comment:
consider renaming to `primitive`, cannot suggest on that line

```suggestion
                let primitive = primitive.to_ident();
                syn::parse_quote_spanned!(Span::call_site() => #primitive)
```